src/Config.php

```diff
@@ -0,0 +1,135 @@
+        return new self(
+            $defaultTimeout,
+            $logger,
+            $commandWorkers,
+            $sessionWorkers,
+            $pruneInterval
+        );
+    }
+
+    /**
+     * @return int The default timeout.
+     */
+    public function defaultTimeout(): int
+    {
+        return $this->defaultTimeout;
+    }
+
+    /**
+     * @return LoggerInterface Defines a target for all the logs.
+     */
+    public function logger (): LoggerInterface
+    {
+        return $this->logger;
+    }
+
+    /**
+     * @return int Maximum accepted command requests.
+     */
+    public function commandWorkers(): int
+    {
+        return $this->commandWorkers;
+    }
+
+    /**
+     * @return int Maximum allowed command responses.
+     */
+    public function sessionWorkers(): int
+    {
+        return $this->sessionWorkers;
+    }
+
+    /**
+     * @return int How often session info is purged.
+     */
+    public function pruneInterval(): int
+    {
+        return $this->pruneInterval;
+    }
+
+
+    /**
+     * @param int             $defaultTimeout The default timeout.
+     * @param LoggerInterface $logger         Defines a target for all the logs.
+     * @param int             $commandWorkers Maximum accepted command requests.
+     * @param int             $sessionWorkers Maximum allowed command responses.
+     * @param int             $pruneInterval  How often session info is purged.
+     */
+    private function __construct(
+        int $defaultTimeout,
+        LoggerInterface $logger,
+        int $commandWorkers,
+        int $sessionWorkers,
+        int $pruneInterval
+    ) {
+        $this->defaultTimeout = $defaultTimeout;
+        $this->logger = $logger;
+        $this->commandWorkers = $commandWorkers;
+        $this->sessionWorkers = $sessionWorkers;
+        $this->pruneInterval = $pruneInterval;
+    }
+
+    /**
+     * DefaultTimeout specifies the maximum amount of time to wait for a call to
+	 * return. It is used if the context passed to Session.call() does not
+	 * already have a deadline.
+     *
+     * @var int The default timeout.
+     */
+    private $defaultTimeout;
+
+    /**
+     * @var LoggerInterface Defines a target for all the logs.
+     */
+    private $logger;
+
+    /**
+     * CommandWorkers is the number of incoming command REQUESTS that are
+	 * accepted at any given time. A new routine is started to service each
+	 * command request.
+     *
+     * @var int Maximum accepted command requests.
+     */
+    private $commandWorkers;
+
+    /**
+     * SessionWorkers is the number of command RESPONSES or notifications that
+     * are buffered in memory at any given time.
+     *
+     * @var int Maximum allowed command responses.
+     */
+    private $sessionWorkers;
+
+    /**
+     * PruneInterval specifies how often the cache of remote session information
+	 * is purged of unused data.
+     * @var How often session info is purged.
+     */
+    private $pruneInterval;
```

Comment:
As discussed previously, we'll need `float` for timeouts and I think we should just roll with `float` for all intervals like this, for consistency.